shop/src/main/java/ro/msg/learning/shop/services/ProductService.java

```diff
@@ -0,0 +1,102 @@
+    private final ProductRepository productRepository;
+    private final ProductMapper productMapper;
+    private final ProductCategoryRepository productCategoryRepository;
+
+    public ProductDTO createProduct(ProductDTO productDto) {
+        Product product = Product.builder()
+                .name(productDto.getName())
+                .description(productDto.getDescription())
+                .price(productDto.getPrice())
+                .weight(productDto.getWeight())
+                .imageUrl(productDto.getImageUrl())
+                .productCategory(isCategory(productDto.getProductCategory()))
+                .build();
+        productRepository.save(product);
+        return productMapper.productToProductDTO(product);
+    }
+
+    public ProductCategory isCategory(ProductCategoryDTO productCategoryDTO) {
+        Optional<ProductCategory> searchedCategory = productCategoryRepository.findByName(productCategoryDTO.getName());
+        ProductCategory productCategory;
+        if (searchedCategory.isPresent()) {
+            productCategory = searchedCategory.get();
+        } else {
+            productCategory = new ProductCategory();
+            productCategory.setName(productCategoryDTO.getName());
+            productCategory.setDescription(productCategoryDTO.getDescription());
+            productCategoryRepository.save(productCategory);
+        }
+        return productCategory;
+
+    }
+
+    public ProductDTO updateProduct(Integer id, ProductDTO updatedProduct) {
+        ProductDTO resultedProduct = null;
+        Optional<Product> productToUpdate = productRepository.findById(id);
+        if (productToUpdate.isPresent()) {
+            Product updated = productToUpdate.get();
+            updated.setName(updatedProduct.getName());
+            updated.setPrice(updatedProduct.getPrice());
+            updated.setDescription(updatedProduct.getDescription());
+            updated.setWeight(updatedProduct.getWeight());
+            updated.setImageUrl(updatedProduct.getImageUrl());
+            updated.setProductCategory(isCategory(updatedProduct.getProductCategory()));
+            productRepository.save(updated);
+            resultedProduct = productMapper.productToProductDTO(updated);
+        }
+        return resultedProduct;
+    }
+
+    public void deleteProductById(Integer id) {
+        productRepository.deleteById(id);
+    }
+
+    public List<ProductDTO> getAllProducts() {
+        List<ProductDTO> existingProducts = new ArrayList<>();
+        try {
+            List<Product> products = productRepository.findAll();
+            if (products.isEmpty()) {
+                throw new ProductNotFoundException("Products not found!");
+            } else {
+                for (Product p : products) {
+                    existingProducts.add(productMapper.productToProductDTO(p));
+                }
+            }
+        } catch (ProductNotFoundException ex) {
+            log.info(ex.getMessage());
```

Comment:
sometimes you catch the custom exception and just log the message, sometimes you throw it and sometimes you just give null back? catching and loging is not good practice as the front end will not know what hit them and why there are no products. Throwing and logging will let both sides know what the issue is. In any way, be consistent with the approach